src/chainstate/stacks/db/transactions.rs

```diff
@@ -442,41 +460,54 @@ impl StacksChainState {
+                        let receipt = StacksTransactionReceipt {
+                            transaction: tx.clone(),
+                            events: vec![],
+                            result: Value::err_none(),
+                            stx_burned: 0
+                        };
+                
                         // abort now -- no burns
-                        return Ok(0);
+                        return Ok(receipt);
                     }
                 };
 
                 // execution -- if this fails due to a runtime error, then the transaction is still
                 // accepted, but the contract does not materialize (but the sender is out their fee).
-                let asset_map = match clarity_tx.connection().initialize_smart_contract(
+                let (asset_map, events) = match clarity_tx.connection().initialize_smart_contract(
                     &contract_id, &contract_ast, &contract_code_str,
                     |asset_map, _| { !StacksChainState::check_transaction_postconditions(&tx.post_conditions, &tx.post_condition_mode, origin_account, asset_map) }) {
-                    Ok(asset_map) => {
-                        Ok(asset_map)
+                    Ok((asset_map, events)) => {
+                        Ok((asset_map, events))
                     },
                     Err(e) => {
                         match e {
                             // runtime errors are okay -- we just have an empty asset map
                             clarity_error::Interpreter(InterpreterError::Runtime(ref runtime_error, ref stack)) => {
-                                debug!("Runtime error {:?} on instantiating {:?}, code {:?}, stack trace {:?}", runtime_error, &contract_id, &contract_code_str, stack);
-                                Ok(AssetMap::new())
+                                info!("Runtime error {:?} on instantiating {}, code {:?}, stack trace {:?}", runtime_error, &contract_id, &contract_code_str, stack);
+                                Ok((AssetMap::new(), vec![]))
                             },
                             _ => Err(e)
                         }
                     }
                 }.map_err(|e| {
-                    warn!("Invalid smart-contract transaction {}: {:?}", &tx.txid(), &e);
+                    info!("Invalid smart-contract transaction {}: {}", &tx.txid(), &e);
                     Error::ClarityError(e)
                 })?;
                 
                 // store analysis -- if this fails, then the have some pretty bad problems
                 clarity_tx.connection().save_analysis(&contract_id, &contract_analysis)
                     .expect("FATAL: failed to store contract analysis");
                 
-                Ok(asset_map.get_stx_burned_total())
+                let receipt = StacksTransactionReceipt {
+                    transaction: tx.clone(),
+                    events,
+                    result: Value::okay_true(),
```

Comment:
Shouldn't `result` here be the final value returned by any "bare code" in the contract?  For example, if my contract is just a `(begin ...)` statement (which is allowed), should the `result` capture the value it evaluates to?

Reply:
hmm good question. I don't have a strong opinion, I personally see the fee `(begin)` statement as a constructor function, which simply returns true when everything went well.
Any take @kantai ?